ravenframework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -71,6 +71,32 @@ def __init__(self):
     self.printTag = 'SyntheticHistoryROM'
     self._dynamicHandling = True # This ROM is able to manage the time-series on its own.
 
+  def __getstate__(self):
+    """
+      Obtains state of object for pickling.
+      @ In, None
+      @ Out, d, dict, stateful dictionary
+    """
+    d = super().__getstate__()
+    d.pop("_saveResiduals", None)
+    d.pop("_residuals", None)
+    d.pop("_globalResiduals", None)
+    return d
+
+  def __setstate__(self, d: dict):
+    """
+      Sets state of object from pickling.
+      @ In, d, dict, stateful dictionary
+      @ Out, None
+    """
+    # For backwards compatibility with existing pickled SyntheticHistory ROMs, there are a few class attributes
+    # to look for. If they're not in the dict coming in for unpickling, we can add them in here before
+    # reconstructing the object.
+    d.setdefault("_saveResiduals", False)
```

Comment:
There probably isn't anyway to avoid hard coding these attributes huh? We couldn't loop through instance attributes and if not in the class attributes add them?

Reply:
Yeah the whole problem is that the attributes just aren't there, so we have to add them. I can't think of a way to do that without hardcoding them.